src/online2/online-nnet2-feature-pipeline.h

```diff
@@ -265,19 +283,33 @@ class OnlineNnet2FeaturePipeline: public OnlineFeatureInterface {
 
   virtual ~OnlineNnet2FeaturePipeline();
  private:
+  /// The following constructor is used internally in the New() function;
+  /// it has the same effect as initializing from just "cfg", but avoids
+  /// re-reading the LDA transform from disk.
+  OnlineNnet2FeaturePipeline(const OnlineNnet2FeaturePipelineInfo &info_,
+                        const Matrix<BaseFloat> &lda_mat,
+                        const Matrix<BaseFloat> &global_cmvn_stats);
 
   const OnlineNnet2FeaturePipelineInfo &info_;
+  /// Init() is to be called from the constructor; it assumes the pointer
+  /// members are all uninitialized but config_ and lda_mat_ are
+  /// initialized.
+  void Init();
```

Comment:
This function was never used or implemented.

Reply:
Thanks! Now it makes more sense, why it did not work :P

Anyway, why is there the `final_feature_` if it is not used by the nnet3 codebase? 
Is it to enforce same number of frames in both feature streams?
K.